In [1862]:
import pandas as pd
import numpy as np

In [1863]:
df = pd.read_csv(r'C:\Users\cleme\laboratorios\PROYECTO-SHARKS-CARLOS\data\attacks.csv', encoding= "ISO-8859-1")

In [1864]:
pd.options.display.max_columns = None


In [1865]:
df.head(5)


Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                                      Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and paddle damaged           N   
1                                 Minor injury to left thigh           N   
2               Injury to left lower leg from surfboard skeg           N   
3                                  Minor injury to lower leg           N   
4          Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00              NaN  K.McMurray, TrackingSharks.com   
2          07h45              NaN  K.McMurray, TrackingSharks.com   
3            NaN        2 m shark                  B. Myatt, GSAF   
4            NaN  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                                                     href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

                                                                             href  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

  Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
0    2018.06.25    2018.06.25          6303.0         NaN         NaN  
1    2018.06.18    2018.06.18          6302.0         NaN         NaN  
2    2018.06.09    2018.06.09          6301.0         NaN         NaN  
3    2018.06.08    2018.06.08          6300.0         NaN         NaN  
4    2018.06.04    2018.06.04          6299.0         NaN         NaN

In [1866]:
df.shape

(25723, 24)

In [1867]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25723 entries, 0 to 25722
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             8702 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6300 non-null   float64
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     5737 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null 

  1 Paso.- Consiste en filtrar todas las filas que contengan un número de celdas sin información en un número de 8 o más y eliminarlas considerando que la tabla tiene 23 columnas. 

In [1868]:
df = df[df.count(axis=1) >= 8]


In [1869]:
df.shape

(6302, 24)

2 Paso. Rellenar todas las celdas sin datos con 'unknow', considerando como 8 celdas en la fila la menor información posible para ser interpretable.

In [1870]:
df = df.fillna('unknown') # rellenar todas las casillas nulas con unknow a fin de limpiar los nulos.

In [1871]:
print(df.head(20))

     Case Number                  Date    Year          Type       Country  \
0     2018.06.25           25-Jun-2018  2018.0       Boating           USA   
1     2018.06.18           18-Jun-2018  2018.0    Unprovoked           USA   
2     2018.06.09           09-Jun-2018  2018.0       Invalid           USA   
3     2018.06.08           08-Jun-2018  2018.0    Unprovoked     AUSTRALIA   
4     2018.06.04           04-Jun-2018  2018.0      Provoked        MEXICO   
5   2018.06.03.b           03-Jun-2018  2018.0    Unprovoked     AUSTRALIA   
6   2018.06.03.a           03-Jun-2018  2018.0    Unprovoked        BRAZIL   
7     2018.05.27           27-May-2018  2018.0    Unprovoked           USA   
8   2018.05.26.b           26-May-2018  2018.0    Unprovoked           USA   
9   2018.05.26.a           26-May-2018  2018.0    Unprovoked           USA   
10    2018.05.24           24-May-2018  2018.0      Provoked     AUSTRALIA   
11    2018.05.21           21-May-2018  2018.0    Unprovoked    

3 paso Elimino la fila de año para rellenarla con el índice de 1 a n registros, en definitiva se denomina "Case number". Ejecuto un bucle  que me cambia esta columna de forma inversa del 1 en la última posición hasta el primero en la posición 6302


In [1872]:

n_fila = df.shape[0] #cambio la lista de sentido, de abajo a arriba.

for i in range(n_fila):
    df.loc[i, 'Case Number'] = n_fila - i


display(df.head(5))
    




Case Number         Date    Year        Type    Country             Area  \
0        6302  25-Jun-2018  2018.0     Boating        USA       California   
1        6301  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2        6300  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3        6299  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4        6298  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex       Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F       57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F       11   
2                    Habush, Oahu      Surfing      John Denges    M       48   
3              Arrawarra Headland      Surfing             male    M  unknown   
4                        La Ticla  Free diving   Gustavo Ramos     M  unknown   

                                                      Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and paddle damaged           N   
1                                 Minor injury to left thigh           N   
2               Injury to left lower leg from surfboard skeg           N   
3                                  Minor injury to lower leg           N   
4          Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00          unknown  K.McMurray, TrackingSharks.com   
2          07h45          unknown  K.McMurray, TrackingSharks.com   
3        unknown        2 m shark                  B. Myatt, GSAF   
4        unknown  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                                                     href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

                                                                             href  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

  Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
0    2018.06.25    2018.06.25          6303.0     unknown     unknown  
1    2018.06.18    2018.06.18          6302.0     unknown     unknown  
2    2018.06.09    2018.06.09          6301.0     unknown     unknown  
3    2018.06.08    2018.06.08          6300.0     unknown     unknown  
4    2018.06.04    2018.06.04          6299.0     unknown     unknown

4 paso Cambio la columna year a integer, para ello, busco celdas q no sean float y encuentro 2 unknow, las cambio y ejecuto y el resultado es el año sin comas.

In [1873]:
resultados = df[df['Year'] == 'unknown'] #localizo unknow.
resultados

Case Number                  Date     Year        Type    Country  \
187         6115  Reported 08-Jan-2017  unknown     Invalid  AUSTRALIA   
6079         223  Reported 19-Aug-1836  unknown  Unprovoked    ENGLAND   

            Area    Location      Activity          Name Sex       Age  \
187   Queensland     unknown  Spearfishing  Kerry Daniel    M       35   
6079  Cumberland  Whitehaven      Swimming         a boy    M  unknown   

                                      Injury Fatal (Y/N)     Time    Species   \
187   No attack, shark made a threat display     unknown  unknown  Bull shark   
6079                                   FATAL           Y  unknown     unknown   

      Investigator or Source                           pdf  \
187   Liquid Vision 1/8/2017  2017.01.08.R-KerryDaniel.pdf   
6079          C. Moore, GSAF   1835.08.19.R-Whitehaven.pdf   

                                                                            href formula  \
187   http://sharkattackfile.net/spreadsheets/pdf_directory/2017.01.08.R-KerryDaniel.pdf   
6079   http://sharkattackfile.net/spreadsheets/pdf_directory/1835.08.19.R-Whitehaven.pdf   

                                                                                    href  \
187   http://sharkattackfile.net/spreadsheets/pdf_directory/2017.01.08.R-KerryDaniel.pdf   
6079   http://sharkattackfile.net/spreadsheets/pdf_directory/1835.08.19.R-Whitehaven.pdf   

     Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
187   2017.01.08.R  2017.01.08.R          6116.0     unknown     unknown  
6079  1836.08.19.R  1836.08.19.R           224.0     unknown     unknown

In [1874]:
df.loc[187,'Year'] = 2017 #cambio al el valor

In [1875]:
df.loc[6079,'Year'] = 1836 #cambio a 0 el valor

In [1876]:
df [df.Year == 'unknown']

Empty DataFrame
Columns: [Case Number, Date, Year, Type, Country, Area, Location, Activity, Name, Sex , Age, Injury, Fatal (Y/N), Time, Species , Investigator or Source, pdf, href formula, href, Case Number.1, Case Number.2, original order, Unnamed: 22, Unnamed: 23]
Index: []

In [1877]:
df['Year'] = df['Year'].astype(int) # paso el type de la columna Year a integer

In [1878]:
display(df.head())  # el retorno es sin coma en 'Year'

Case Number         Date  Year        Type    Country             Area  \
0        6302  25-Jun-2018  2018     Boating        USA       California   
1        6301  18-Jun-2018  2018  Unprovoked        USA          Georgia   
2        6300  09-Jun-2018  2018     Invalid        USA           Hawaii   
3        6299  08-Jun-2018  2018  Unprovoked  AUSTRALIA  New South Wales   
4        6298  04-Jun-2018  2018    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex       Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F       57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F       11   
2                    Habush, Oahu      Surfing      John Denges    M       48   
3              Arrawarra Headland      Surfing             male    M  unknown   
4                        La Ticla  Free diving   Gustavo Ramos     M  unknown   

                                                      Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and paddle damaged           N   
1                                 Minor injury to left thigh           N   
2               Injury to left lower leg from surfboard skeg           N   
3                                  Minor injury to lower leg           N   
4          Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00          unknown  K.McMurray, TrackingSharks.com   
2          07h45          unknown  K.McMurray, TrackingSharks.com   
3        unknown        2 m shark                  B. Myatt, GSAF   
4        unknown  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                                                     href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

                                                                             href  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

  Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
0    2018.06.25    2018.06.25          6303.0     unknown     unknown  
1    2018.06.18    2018.06.18          6302.0     unknown     unknown  
2    2018.06.09    2018.06.09          6301.0     unknown     unknown  
3    2018.06.08    2018.06.08          6300.0     unknown     unknown  
4    2018.06.04    2018.06.04          6299.0     unknown     unknown

paso 5. Cambio los valores de la columna género a m y f

In [1879]:
print(df['Sex '].value_counts()) # reviso cuantos tipos de genero hay, para ello muestro las denominaciones de valores que hay en la columna

M          5094
F           637
unknown     565
M             2
N             2
lli           1
.             1
Name: Sex , dtype: int64


In [1880]:
resultados_1= df[df['Sex '] == '.'] #Investigo que hay en las columnas, si contienen información valiosa
resultados_1

Case Number                  Date  Year          Type           Country  \
5437         865  Reported 02-Jun-1908  1908  Sea Disaster  PAPUA NEW GUINEA   

             Area Location Activity     Name Sex       Age  \
5437  New Britain   Matupi        .  unknown    .  unknown   

                                                                                          Injury  \
5437  Remains of 3 humans recovered from shark, but shark involvement prior to death unconfirmed   

     Fatal (Y/N)     Time                   Species   \
5437           Y  unknown  Allegedly a 33-foot shark   

         Investigator or Source                      pdf  \
5437  Taranaki Herald, 6/2/1908  1908.06.02.R-Matupi.pdf   

                                                                       href formula  \
5437  http://sharkattackfile.net/spreadsheets/pdf_directory/1908.06.02.R-Matupi.pdf   

                                                                               href  \
5437  http://sharkattackfile.net/spreadsheets/pdf_directory/1908.06.02.R-Matupi.pdf   

     Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
5437  1908.06.02.R  1908.06.02.R           866.0     unknown     unknown

In [1881]:
resultados_1= df[df['Sex '] == 'lli']
resultados_1

Case Number         Date  Year        Type Country        Area  \
1624        4678  11-Nov-2004  2004  Unprovoked     USA  California   

                                            Location Activity        Name  \
1624  Bunkers, Humboldt Bay, Eureka, Humboldt County  Surfing  Brian Kang   

     Sex  Age                              Injury Fatal (Y/N)   Time  \
1624  lli  38  Lacerations to hand, knee & thigh            N  13h30   

                     Species             Investigator or Source  \
1624  5.5 m [18'] white shark  R. Collier, GSAF                   

                        pdf  \
1624  2004.11.11.b-Kang.pdf   

                                                                     href formula  \
1624  http://sharkattackfile.net/spreadsheets/pdf_directory/2004.11.11.b-Kang.pdf   

                                                                             href  \
1624  http://sharkattackfile.net/spreadsheets/pdf_directory/2004.11.11.b-Kang.pdf   

     Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
1624  2004.11.11.b  2004.11.11.b          4679.0     unknown     unknown

In [1882]:
resultados_1= df[df['Sex '] == 'N']
resultados_1

Case Number                  Date  Year      Type    Country  \
4938        1364           11-Jul-1934  1934   Boating  AUSTRALIA   
6131         171  Reported 18-Dec-1801  1801  Provoked    unknown   

                 Area  Location                         Activity  \
4938  New South Wales  Cronulla                          Fishing   
6131          unknown   unknown  Standing on landed shark's tail   

                                             Name Sex       Age  \
4938  18' boat, occupants William & Leslie Newton    N  unknown   
6131                             Stephen Pettigew    N  unknown   

                                                                                          Injury  \
4938  No injury to occupants Sharks continually followed the dinghy, and one smashed its rudder    
6131                                                                    FATAL, PROVOKED INCIDENT   

     Fatal (Y/N)     Time            Species   \
4938           N  unknown  Blue pointer, 11'    
6131           Y  unknown           12' shark   

                                                                Investigator or Source  \
4938  G.P. Whitley, ref: Daily Telegraph, 7/11/1934 & Sydney Morning Herald  7/12/1934   
6131                                                      The Evening Post, 12/18/1801   

                                       pdf  \
4938  1934.07.11-Newton-boat-Australia.pdf   
6131            1801.12.18.R-Pettigrew.pdf   

                                                                                    href formula  \
4938  http://sharkattackfile.net/spreadsheets/pdf_directory/1934.07.11-Newton-boat-Australia.pdf   
6131            http://sharkattackfile.net/spreadsheets/pdf_directory/1801.12.18.R-Pettigrew.pdf   

                                                                                            href  \
4938  http://sharkattackfile.net/spreadsheets/pdf_directory/1934.07.11-Newton-boat-Australia.pdf   
6131            http://sharkattackfile.net/spreadsheets/pdf_directory/1801.12.18.R-Pettigrew.pdf   

     Case Number.1 Case Number.2  original order Unnamed: 22 Unnamed: 23  
4938    1934.07.11    1934.07.11          1365.0     unknown     unknown  
6131  1801.12.18.R  1801.12.18.R           172.0     unknown     unknown

In [1883]:
df['Sex '].replace(['lli'], ['M'], inplace=True) #reemplazo 'lli' por 'M'

In [1884]:
df['Sex '].replace(['.'], ['unknown'], inplace=True) #reemplazo '.' por 'unknow'

In [1885]:
df['Sex '].replace(['N'], ['unknown'], inplace=True) #reemplazo 'N' por 'M'

In [1886]:
df['Sex '].replace(['M '], ['M'], inplace=True)

In [1887]:
print(df['Sex '].value_counts()) # compruebo que me queda limpia con sólo 3 valores

M          5097
F           637
unknown     568
Name: Sex , dtype: int64


paso 6 Exploro columnas iguales y les cambio el nombre, elimino los resultados de las columnas, si están repetidos o los renombro

In [1888]:
df = df.rename(columns={'href': 'Unvalid3'}) #nuevo nombre de columna

In [1889]:
df['Unvalid3'] = 'See_href_formula' #Cambio los valores por repetidos
print(df['Unvalid3'].value_counts())

See_href_formula    6302
Name: Unvalid3, dtype: int64


In [1890]:
df = df.rename(columns={'Case Number.1': 'Empty1'})#nuevo nombre de columna


In [1891]:
df['Empty1'] = 'no_values' #Cambio los valores por repetidos con Date
print(df['Empty1'].value_counts())

no_values    6302
Name: Empty1, dtype: int64


In [1892]:
df = df.rename(columns={'Case Number.2': 'Empty2'})#nuevo nombre de columna


In [1893]:
df['Empty2'] = 'no_values' #Cambio los valores por repetidos con Date
print(df['Empty2'].value_counts())

no_values    6302
Name: Empty2, dtype: int64


In [1894]:
df['original order'] = df['original order'].astype(str) #cambio de float a string


In [1895]:
df = df.rename(columns={'original order': 'Empty3'})#nuevo nombre de columna

In [1896]:
df['Empty3'] = 'no_values' #Cambio los valores por repetidos con Case number
print(df['Empty3'].value_counts())

no_values    6302
Name: Empty3, dtype: int64


In [1897]:
df.head()

Case Number         Date  Year        Type    Country             Area  \
0        6302  25-Jun-2018  2018     Boating        USA       California   
1        6301  18-Jun-2018  2018  Unprovoked        USA          Georgia   
2        6300  09-Jun-2018  2018     Invalid        USA           Hawaii   
3        6299  08-Jun-2018  2018  Unprovoked  AUSTRALIA  New South Wales   
4        6298  04-Jun-2018  2018    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex       Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F       57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F       11   
2                    Habush, Oahu      Surfing      John Denges    M       48   
3              Arrawarra Headland      Surfing             male    M  unknown   
4                        La Ticla  Free diving   Gustavo Ramos     M  unknown   

                                                      Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and paddle damaged           N   
1                                 Minor injury to left thigh           N   
2               Injury to left lower leg from surfboard skeg           N   
3                                  Minor injury to lower leg           N   
4          Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00          unknown  K.McMurray, TrackingSharks.com   
2          07h45          unknown  K.McMurray, TrackingSharks.com   
3        unknown        2 m shark                  B. Myatt, GSAF   
4        unknown  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                                                     href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

           Unvalid3     Empty1     Empty2     Empty3 Unnamed: 22 Unnamed: 23  
0  See_href_formula  no_values  no_values  no_values     unknown     unknown  
1  See_href_formula  no_values  no_values  no_values     unknown     unknown  
2  See_href_formula  no_values  no_values  no_values     unknown     unknown  
3  See_href_formula  no_values  no_values  no_values     unknown     unknown  
4  See_href_formula  no_values  no_values  no_values     unknown     unknown

Paso 7 Cambiar tipo columna 'Date' a datetime para su correcto manejo futuro

In [1898]:
tipo_columna = df['Date'].dtype # Verifico el tipo object
print(tipo_columna)


object


In [1899]:
from datetime import date #importo librerias
from datetime import datetime

In [1900]:
df['Date'] = pd.to_datetime(df['Date'],errors='coerce') #cambio de un tipo a otro
df.head()


Case Number       Date  Year        Type    Country             Area  \
0        6302 2018-06-25  2018     Boating        USA       California   
1        6301 2018-06-18  2018  Unprovoked        USA          Georgia   
2        6300 2018-06-09  2018     Invalid        USA           Hawaii   
3        6299 2018-06-08  2018  Unprovoked  AUSTRALIA  New South Wales   
4        6298 2018-06-04  2018    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex       Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F       57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F       11   
2                    Habush, Oahu      Surfing      John Denges    M       48   
3              Arrawarra Headland      Surfing             male    M  unknown   
4                        La Ticla  Free diving   Gustavo Ramos     M  unknown   

                                                      Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and paddle damaged           N   
1                                 Minor injury to left thigh           N   
2               Injury to left lower leg from surfboard skeg           N   
3                                  Minor injury to lower leg           N   
4          Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00          unknown  K.McMurray, TrackingSharks.com   
2          07h45          unknown  K.McMurray, TrackingSharks.com   
3        unknown        2 m shark                  B. Myatt, GSAF   
4        unknown  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                                                     href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

           Unvalid3     Empty1     Empty2     Empty3 Unnamed: 22 Unnamed: 23  
0  See_href_formula  no_values  no_values  no_values     unknown     unknown  
1  See_href_formula  no_values  no_values  no_values     unknown     unknown  
2  See_href_formula  no_values  no_values  no_values     unknown     unknown  
3  See_href_formula  no_values  no_values  no_values     unknown     unknown  
4  See_href_formula  no_values  no_values  no_values     unknown     unknown

paso 8 Agrupo valores en Y,N, unknown. quito valores no válidos para busqueda

In [1901]:
print(df['Fatal (Y/N)'].value_counts())

N          4293
Y          1388
unknown     539
UNKNOWN      71
 N            7
M             1
2017          1
N             1
y             1
Name: Fatal (Y/N), dtype: int64


In [1902]:
df['Fatal (Y/N)'].replace(['UNKNOWN','2017', 'M'], ['unknown', 'unknown', 'unknown'], inplace=True) #reemplazo valores dudosos por 'unknow'


In [1903]:
df['Fatal (Y/N)'].replace([' N','N ', 'y'], ['N', 'N', 'Y'], inplace=True) #reemplazo valores dudosos por 'unknow'

In [1904]:
print(df['Fatal (Y/N)'].value_counts())

N          4301
Y          1389
unknown     612
Name: Fatal (Y/N), dtype: int64


In [1905]:
df.head()

Case Number       Date  Year        Type    Country             Area  \
0        6302 2018-06-25  2018     Boating        USA       California   
1        6301 2018-06-18  2018  Unprovoked        USA          Georgia   
2        6300 2018-06-09  2018     Invalid        USA           Hawaii   
3        6299 2018-06-08  2018  Unprovoked  AUSTRALIA  New South Wales   
4        6298 2018-06-04  2018    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex       Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F       57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F       11   
2                    Habush, Oahu      Surfing      John Denges    M       48   
3              Arrawarra Headland      Surfing             male    M  unknown   
4                        La Ticla  Free diving   Gustavo Ramos     M  unknown   

                                                      Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and paddle damaged           N   
1                                 Minor injury to left thigh           N   
2               Injury to left lower leg from surfboard skeg           N   
3                                  Minor injury to lower leg           N   
4          Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00          unknown  K.McMurray, TrackingSharks.com   
2          07h45          unknown  K.McMurray, TrackingSharks.com   
3        unknown        2 m shark                  B. Myatt, GSAF   
4        unknown  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                                                     href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.25-Wolfe.pdf   
1    http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.18-McNeely.pdf   
2     http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.09-Denges.pdf   
3  http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.08-Arrawarra.pdf   
4      http://sharkattackfile.net/spreadsheets/pdf_directory/2018.06.04-Ramos.pdf   

           Unvalid3     Empty1     Empty2     Empty3 Unnamed: 22 Unnamed: 23  
0  See_href_formula  no_values  no_values  no_values     unknown     unknown  
1  See_href_formula  no_values  no_values  no_values     unknown     unknown  
2  See_href_formula  no_values  no_values  no_values     unknown     unknown  
3  See_href_formula  no_values  no_values  no_values     unknown     unknown  
4  See_href_formula  no_values  no_values  no_values     unknown     unknown

paso 9 Limpiar y unificar los tipos de accidente en la columna 'Type'

In [1906]:
print(df['Type'].value_counts())

Unprovoked      4595
Provoked         574
Invalid          547
Sea Disaster     239
Boating          203
Boat             137
unknown            4
Questionable       2
Boatomg            1
Name: Type, dtype: int64


In [1907]:
df['Type'].replace(['Boat','Boatomg', 'Questionable', ], ['Boating', 'Boating', 'Invalid'], inplace=True) #reagrupo valores equivalentes

In [1908]:
print(df['Type'].value_counts())

Unprovoked      4595
Provoked         574
Invalid          549
Boating          341
Sea Disaster     239
unknown            4
Name: Type, dtype: int64
